quickwit-metastore/src/metastore_resolver.rs

```diff
@@ -0,0 +1,82 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use crate::{Metastore, MetastoreResolverError};
+
+/// A metastore factory builds a [`Metastore`] object from an URI.
+#[cfg_attr(any(test, feature = "testsuite"), mockall::automock)]
```

Comment:
It might be worth highlighting the intent in the comment here, or put some examples.

At the moment, `SingleFileMetaStore` is the only implementation available, so a metastore uri is simply the index storage uri. (BTW do we add metastore.json to that?)
In the future, we will handle different backend (e.g. postgresql) that will also be identified by an URI.